packages/babel-traverse/src/path/inference/inferer-reference.js

```diff
@@ -99,6 +99,10 @@ function getTypeAnnotationBindingConstantViolations(binding, path, name) {
     return t.createTSUnionType(types);
   }
 
+  if (t.createFlowUnionType) {
```

Comment:
Check if it's available (in new versions only)
Old version will go further

The same for other cases below